src/d3a/d3a_core/export.py

```diff
@@ -86,13 +88,35 @@ def __init__(self, root_area: Area, path: str, subdir: str, endpoint_buffer):
             _log.error("Could not open directory for csv exports: %s" % str(ex))
             return
 
-    def export_json_data(self, directory: dir, area: Area):
+    def generate_json_report(self):
```

Comment:
I would very much prefer this method to stay in the endpoint_buffer class. Reason for that is that this method only depends on the endpoint_buffer, therefore you can replace all the `self.endpoint_buffer` to just `self`. In general, this is usually a very good indication that a method is a better fit to another class.